core/templates/pages/landing-pages/topic-landing-page/topic-landing-page.component.spec.ts

```diff
@@ -16,96 +16,128 @@
-    expect(ctrl.topicTitle).toBe('Ratios');
+
+  it('should get information from topic identified at pathname', () => {
+    windowRef.nativeWindow.location.pathname = '/math/ratios';
+    component.ngOnInit();
+    expect(component.topicTitle).toBe('Ratios');
   });
 
-  it('should click get started button', function() {
-    var nativeWindowSpy = spyOnProperty(windowRef, 'nativeWindow');
-    nativeWindowSpy.and.returnValue({
-      location: {
-        pathname: '/math/ratios'
-      }
-    });
+  it('should click get started button', fakeAsync(() => {
+    windowRef.nativeWindow.location.pathname = '/math/ratios';
     var analyticsSpy = spyOn(
-      SiteAnalyticsService, 'registerOpenCollectionFromLandingPageEvent')
+      siteAnalyticsService, 'registerOpenCollectionFromLandingPageEvent')
       .and.callThrough();
     // Get collection id from ratios.
-    ctrl.$onInit();
+    component.ngOnInit();
 
-    nativeWindowSpy.and.returnValue({
-      location: ''
-    });
-    ctrl.onClickGetStartedButton();
+    windowRef.nativeWindow.location.href = '';
+    component.onClickGetStartedButton();
 
     var ratiosCollectionId = '53gXGLIR044l';
     expect(analyticsSpy).toHaveBeenCalledWith(ratiosCollectionId);
-    $timeout.flush(150);
+    tick(150);
+    fixture.detectChanges();
 
-    expect(windowRef.nativeWindow.location).toBe(
+    expect(windowRef.nativeWindow.location.href).toBe(
       '/collection/' + ratiosCollectionId);
-  });
+  }));
 
-  it('should click learn more button', function() {
-    spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
-      location: ''
-    });
-    ctrl.onClickLearnMoreButton();
-    $timeout.flush(150);
+  it('should click learn more button', fakeAsync(() => {
+    windowRef.nativeWindow.location.href = '';
+    component.onClickLearnMoreButton();
+    tick(150);
+    fixture.detectChanges();
 
-    expect(windowRef.nativeWindow.location).toBe('/community-library');
-  });
+    expect(windowRef.nativeWindow.location.href).toBe('/community-library');
+  }));
 
-  it('should have a tagline in the page title', function() {
-    spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
-      location: {
-        pathname: '/math/fractions'
-      }
-    });
-    ctrl.$onInit();
-    $timeout(() => {
-      expect($window.document.title).toBe('Fractions | ' +
-        'Add, Subtract, Multiply and Divide | Oppia');
-    }, 150, false);
-  });
+  it('should have a tagline in the page title', fakeAsync(() => {
+    windowRef.nativeWindow.location.pathname = '/math/fractions';
+    component.ngOnInit();
+    tick(150);
+    fixture.detectChanges();
+    expect(pageTitleService.getPageTitle()).toBe('Fractions | ' +
+      'Add, Subtract, Multiply and Divide | Oppia');
+  }));
 
   it('should return correct lesson quality image src', function() {
-    var imageSrc = ctrl.getLessonQualityImageSrc('someImage.png');
+    var imageSrc = component.getLessonQualityImageSrc('someImage.png');
```

Comment:
```suggestion
    const imageSrc = component.getLessonQualityImageSrc('someImage.png');
```

Reply:
Changed to let, since the same variable was being assigned again in the test.